DifferentiationInterface/ext/DifferentiationInterfaceMooncakeExt/differentiate_with.jl

```diff
@@ -0,0 +1,48 @@
+@is_primitive MinimalCtx Tuple{DI.DifferentiateWith,<:Union{Number,AbstractArray}}
+
+function Mooncake.rrule!!(dw::CoDual{<:DI.DifferentiateWith}, x::CoDual{<:Number})
+    primal_func = primal(dw)
+    primal_x = primal(x)
+    (; f, backend) = primal_func
+    y = zero_fcodual(f(primal_x))
+
+    # output is a vector, so we need to use the vector pullback
```

Comment:
I don't think it's a good idea to define two methods for `pullback!!`. We should probably define `pullback_scalar!!` and `pullback_array!!`, then return one of those and only one.